lib/Service/ObjectService.php

```diff
@@ -696,85 +696,92 @@ public function saveObject(int|string|Register $register, int|string|Schema $sch
         }
 
         if ($depth === null) {
-			$depth = $schema->getMaxDepth();
-		}
+            $depth = $schema->getMaxDepth();
+        }
 
-		if (isset($object['id']) === true) {
-			$objectEntity = $this->objectEntityMapper->find(
-				$object['id']
-			);
-		}
-		else {
-			$objectEntity = new ObjectEntity();
-			$objectEntity->setRegister($register->getId());
-			$objectEntity->setSchema($schema->getId());
-		}
+        if (isset($object['id']) === true) {
+            try {
+                $objectEntity = $this->objectEntityMapper->find(
+                    $object['id']
+                );
+            } catch (DoesNotExistException $exception) {
+                $objectEntity = new ObjectEntity();
```

Comment:
Do we want this? Why not just return a does not exist exception here? If you are trying to update an existing object by passing an id we should not be creating a new object if it doesn't exist, right?